src/tree/index.tsx

```diff
@@ -0,0 +1,382 @@
+		}
+
+		const nodes = flat(results);
+
+		return (
+			<ol
+				classes={[classes.root, theme.variant()]}
+				focus={() => shouldFocus}
+				onkeydown={onKeyDown}
+				tabIndex={0}
+			>
+				{nodes.map((node) => {
+					const isExpanded = expandedNodes.indexOf(node.id) !== -1;
+					return (
+						<li
+							classes={[
+								classes.node,
+								node.hasChildren && classes.leaf,
+								selectable && classes.selectable,
+								node.id === selectedNode && classes.selected
+							]}
+						>
+							<TreeNode
+								depth={depth}
+								activeNode={activeNode}
+								checkable={checkable}
+								selectable={selectable}
+								checkedNodes={checkedNodes}
+								selectedNode={selectedNode}
+								disabledNodes={disabledNodes || []}
+								expandedNodes={expandedNodes}
+								node={node}
+								onActive={activateNode}
+								onSelect={selectNode}
+								onCheck={(n, checked) => {
+									if (checked) {
+										checkedNodes.push(n);
+									} else {
+										checkedNodes.splice(checkedNodes.indexOf(n), 1);
+									}
+									icache.set('checkedNodes', checkedNodes);
+									onCheck && onCheck(n, checked);
+								}}
+								onExpand={(n, expanded) => {
+									if (expanded) {
+										expandNode(n);
+									} else {
+										collapseNode(n);
+									}
+								}}
+							>
+								{itemRenderer || defaultRenderer}
+							</TreeNode>
+							{isExpanded && mapNodeTree(node.id, depth + 1)}
+						</li>
+					);
+				})}
+			</ol>
+		);
+	}
+	return mapNodeTree();
+});
+
+/*******************
+ * TreeNode
+ *******************/
+
+interface TreeNodeProperties {
+	checkable: boolean;
+	depth: number;
+	selectable: boolean;
+	activeNode?: string;
+	selectedNode?: string;
+	checkedNodes: string[];
+	disabledNodes: string[];
+	expandedNodes: string[];
+	node: TreeNodeOption;
+	onActive(node: string): void;
+	onSelect(node: string): void;
+	onCheck(node: string, checked: boolean): void;
+	onExpand(node: string, expanded: boolean): void;
+}
+
+interface TreeNodeChildren {
+	(node: TreeNodeOption): RenderResult;
+}
+
+const treeNodeFactory = create({ theme })
+	.properties<TreeNodeProperties>()
+	.children<TreeNodeChildren>();
+
+export const TreeNode = treeNodeFactory(function({ middleware: { theme }, properties, children }) {
+	const {
+		node,
+		checkable,
+		depth,
+		selectable,
+		activeNode,
+		selectedNode,
+		checkedNodes,
+		disabledNodes,
+		expandedNodes,
+		onActive,
+		onSelect,
+		onCheck,
+		onExpand
+	} = properties();
+	const [itemRenderer] = children();
+	const classes = theme.classes(css);
+	const isActive = node.id === activeNode;
+	const isSelected = node.id === selectedNode;
+	const expanded = expandedNodes.includes(node.id);
+	const checked = checkedNodes.includes(node.id);
+	const isDisabled = disabledNodes && disabledNodes.includes(node.id);
+	const isExpandable = node.hasChildren;
+	const spacers = new Array(depth);
+	fill(spacers, <div classes={classes.spacer} />);
```

Comment:
I think that this should be possible by simply putting a `margin-left`or `padding-left` on each of the nested `ol` nodes. We shouldn't be creating `spacer` divs like this.

Reply:
I believe this was done as the further nested the node are, the more spacing they need. So the styling needed  correlate with the node depth.

Reply:
given that each list will be within a parent list it should work fine